pkg/reconciler/stats_reporter.go

```diff
@@ -176,6 +176,25 @@ type StatsReporter interface {
 	ReportReady(kind, namespace, service string, d time.Duration) error
 }
 
+// srKey is used to associate StatsReporters with contexts.
+type srKey struct{}
+
+// WithStatsReporter attaches the given StatsReporter to the provided context
+// in the returned context.
+func WithStatsReporter(ctx context.Context, sr StatsReporter) context.Context {
+	return context.WithValue(ctx, srKey{}, sr)
+}
+
+// GetStatsReporter attempts to look up the StatsReporter on a given context.
+// It may return null if none is found.
+func GetStatsReporter(ctx context.Context) StatsReporter {
+	untyped := ctx.Value(srKey{})
+	if untyped == nil {
+		return nil
+	}
+	return untyped.(StatsReporter)
+}
```

Comment:
same: move to knative/pkg

Reply:
The stats reporters are not the same.

Reply:
how do you feel about:
```
// StatsReporter reports reconcilers' metrics.
type StatsReporter interface {
	// ReportReady reports the time it took a resource to become Ready.
	ReportReady(kind, namespace, service string, d time.Duration) error
}
```

Reply:
tracking this with an issue: https://github.com/knative/pkg/issues/446